Patch-Source: https://salsa.debian.org/debian/shntool/-/commit/57efcd7b34c2107dd785c11d79dfcd4520b2bc41
--
Origin: https://github.com/max619/shntool/commit/cfd06e4ed
From cfd06e4edecdca2013e0fe04db135fd110a68203 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=D0=9C=D0=B0=D0=BA=D1=81=D0=B8=D0=BC=20=D0=91=D0=B0=D0=B3?=
 =?UTF-8?q?=D1=80=D1=8F=D0=BD=D1=86=D0=B5=D0=B2?= <maxbag97619@gmail.com>
Date: Mon, 11 Nov 2019 00:53:06 +0300
Subject: [PATCH] Added WAVE_FORMAT_EXTENSIBLE with value 0xfffe and supressed
 error on wav header parsing with that format

--- a/include/wave.h
+++ b/include/wave.h
@@ -58,8 +58,9 @@
 #define WAVE_FORMAT_MPEG                (0x0050)
 #define WAVE_FORMAT_MPEGLAYER3          (0x0055)
 #define WAVE_FORMAT_G726_ADPCM          (0x0064)
 #define WAVE_FORMAT_G722_ADPCM          (0x0065)
+#define WAVE_FORMAT_EXTENSIBLE          (0xfffe)
 
 #define CD_BLOCK_SIZE                   (2352)
 #define CD_BLOCKS_PER_SEC               (75)
 #define CD_BLOCK_ALIGN                  (4)
--- a/src/core_wave.c
+++ b/src/core_wave.c
@@ -166,8 +166,9 @@
     return FALSE;
   }
 
   switch (info->wave_format) {
+    case WAVE_FORMAT_EXTENSIBLE:
     case WAVE_FORMAT_PCM:
       break;
     default:
       st_warning("unsupported format 0x%04x (%s) while processing file: [%s]",
@@ -510,9 +511,11 @@
       return "MPEG Layer 3";
     case WAVE_FORMAT_G726_ADPCM:
       return "G.726 ADPCM";
     case WAVE_FORMAT_G722_ADPCM:
-      return "G.722 ADPCM";
+      return "G.722 ADPCM";      
+    case WAVE_FORMAT_EXTENSIBLE:
+      return "WAVE Extensible format";
   }
   return "Unknown";
 }
